### PR TITLE
htmldoc: fix darwin build

### PR DESCRIPTION
**Pull request by jiegec**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/typesetting/htmldoc/default.nix [hunk 1]
```diff
@@ -14,8 +14,14 @@ stdenv.mkDerivation rec {
   buildInputs = [ zlib libpng libjpeg ]
     ++ lib.optionals stdenv.isDarwin [ Foundation SystemConfiguration ];
 
+  # do not generate universal binary on Darwin
+  # because it is not supported by Nix's clang
+  preConfigurePhases = "preConfigure";
+  preConfigure = ''
+    substituteInPlace configure --replace "-arch x86_64 -arch arm64" ""
+  '';
+
```
Comment (NickCao) [inline pkgs/tools/typesetting/htmldoc/default.nix:23]:
```suggestion
  postPatch = ''
    substituteInPlace configure --replace "-arch x86_64 -arch arm64" ""
  '';

```

**Comment by jiegec** [comment]
Result of `nixpkgs-review pr 178725` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>htmldoc</li>
  </ul>
</details>

**Comment by jiegec** [comment]
Currently, htmldoc is broken on x86_64-darwin:

```
       > Compiling string.c...
       > Compiling tls.c...
       > In file included from tls.c:42:
       > ./tls-darwin.c:955:7: error: use of undeclared identifier 'kTLSProtocol13'
       >       kTLSProtocol13
       >       ^
       > 1 error generated.
       > make[1]: *** [../Makedefs:89: tls.o] Error 1
       > make[1]: Leaving directory '/private/tmp/nix-build-htmldoc-1.9.16.drv-0/source/htmldoc'
       > make: *** [Makefile:31: all] Error 1

```

It seems to be related to macOS SDK version, 10.13 is required.

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 178725` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>htmldoc</li>
  </ul>
</details>

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 178725` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>htmldoc</li>
  </ul>
</details>

File: pkgs/tools/typesetting/htmldoc/default.nix [hunk 2]
```diff
@@ -1,11 +1,11 @@
-{ lib, stdenv, fetchFromGitHub, zlib, libpng, libjpeg, SystemConfiguration, Foundation, pkg-config }:
+{ lib, stdenv, testers, fetchFromGitHub, zlib, libpng, libjpeg, SystemConfiguration, Foundation, pkg-config, htmldoc }:
 
 stdenv.mkDerivation rec {
   pname = "htmldoc";
   version = "1.9.16";
   src = fetchFromGitHub {
     owner = "michaelrsweet";
-    repo = "htmldoc";
+    repo = pname;
```
Comment (SuperSandro2000) [inline pkgs/tools/typesetting/htmldoc/default.nix:8]:
```suggestion
    repo = "htmldoc";
```

File: pkgs/tools/typesetting/htmldoc/default.nix [hunk 3]
```diff
@@ -14,8 +14,18 @@ stdenv.mkDerivation rec {
   buildInputs = [ zlib libpng libjpeg ]
     ++ lib.optionals stdenv.isDarwin [ Foundation SystemConfiguration ];
 
+  # do not generate universal binary on Darwin
+  # because it is not supported by Nix's clang
```
Comment (SuperSandro2000) [inline pkgs/tools/typesetting/htmldoc/default.nix:18]: Could this be fixed in nix's clang?

**Pull request merged**